packages/blockchain-wallet-v4-frontend/src/modals/ShowEthPrivateKey/index.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import { compose, bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import modalEnhancer from 'providers/ModalEnhancer'
+import { actions, selectors } from 'data'
+import ShowEthPrivateKeyTemplate from './template'
+
+class ShowEthPrivateKeyContainer extends Component {
+  componentWillUnmount () {
+    this.props.actions.clearShownEthPrivateKey()
+  }
+
+  render () {
+    let step = this.props.priv == null ? 0 : 1
+    let nextStep = () => this.props.actions.showEthPrivateKey(this.props.archived)
+    return (
+      <ShowEthPrivateKeyTemplate
+        {...this.props}
+        step={step}
+        onContinue={nextStep}
+      />
+    )
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
+  const ethKvStoreSelectors = selectors.core.kvStore.ethereum
```

Comment:
This should live in a separate file (selectors.js), to make it easier to unit test